authz/tests/fga_test.go

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright 2023 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package tests runs the fga CLI tests against each file in this directory.
+// It assumes the `fga` binary is in the PATH, e.g. from `make bootstrap`
+package tests
+
+import (
+	"bytes"
+	"path/filepath"
+	"testing"
+
+	// We need the init() function here, but we can't copy rootCmd because it is private
+	_ "github.com/openfga/cli/cmd"
+	"github.com/openfga/cli/cmd/model"
+)
+
+func TestFGA(t *testing.T) {
+	files, err := filepath.Glob("*.tests.yaml")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(files) == 0 {
+		t.Fatal("no test files found")
+	}
+	for _, f := range files {
+		file := f
+		t.Run(file, func(t *testing.T) {
+			// We invoke cobra commands directly, which references some global state in FGA.
+			// Call .SetEnv here to hint to paralleltest that this can't be parallelized.
+			// See https://github.com/kunwardeep/paralleltest/pull/33
+			t.Setenv("PARALLELTEST_NO_PARALLEL", "true")
```

Comment:
I would love to have a comment or something that can quiet `parelleltest`, but the code seems to be constructed to parse the AST for methods, rather than comment-driven.  https://github.com/kunwardeep/paralleltest/pull/33 provides a hook to do this, though it also forces us to make this test not run at all in parallel with our other tests.  It runs in 0.2s on my MacBook, so it's not too bad at the moment.

Reply:
... and `golangci-lint` hasn't been released to pick up that change yet, so I needed to edit the workflow, too.

Reply:
I just figured out that I can just use a file-level mutex around calls to `.Execute()`, which is _much_ simpler.